Defect: SMI-MISDECLARED-ACCESS https://pysnmp.github.io/pysmi/stable/mib-defects.html#smi-misdeclared-access
Defect: SMI-UNDEFINED-NAME https://pysnmp.github.io/pysmi/stable/mib-defects.html#smi-undefined-name
Defect: SMI-MISSING-IMPORT https://pysnmp.github.io/pysmi/stable/mib-defects.html#smi-missing-import

--- a/MIP-MIB
+++ b/MIP-MIB
@@ -2,7 +2,7 @@
 
     IMPORTS
         Counter32, Gauge32, Integer32, IpAddress, experimental,
-        MODULE-IDENTITY, OBJECT-TYPE, NOTIFICATION-TYPE
+        MODULE-IDENTITY, OBJECT-TYPE, NOTIFICATION-TYPE, mib-2
                                         FROM SNMPv2-SMI
         RowStatus, TruthValue, TimeStamp,
         TEXTUAL-CONVENTION
@@ -258,7 +258,7 @@
 
     mipSecViolatorAddress  OBJECT-TYPE
         SYNTAX      IpAddress
-        MAX-ACCESS  accessible-for-notify
+        MAX-ACCESS  read-only
         STATUS      current
         DESCRIPTION
                 "Violator's IP address. The violator is not necessary
@@ -2096,7 +2096,7 @@
                 function within a home agent."
         ::= { mipGroups 12 }
 
-    mipSecNotifcationsGroup NOTIFICATION-GROUP
+    mipSecNotificationsGroup NOTIFICATION-GROUP
         NOTIFICATIONS { mipAuthFailure }
         STATUS      current
         DESCRIPTION
